### ISSUE TITLE
Clean old img sprite

### ISSUE THREAD

**Issue by rustam87** [mixtur/webpack-spritesmith#45]
Have a option clean old sprite when generate new sprite and i must yourself?
Or to disable watch files, because old files not removed
Or use webpack clean plugin when I add/remove images

**Comment by rustam87** [comment]
Do you want to do this feature?  clean old sprite when generate

**Comment by mixtur** [comment]
Yes I will. But I still recomend to use file-loader instead.

**Comment by liorch88** [comment]
This fix has caused a major issue for my configuration: In my webpack config file, I am using the SpritesmithPlugin multiple times, for different sprites (with different targets). This has worked well for me in the previous versions.
What happens now,on this version, is that for the first target it creates a file "A.scss". Then, it gets to the next target and creates file "B.scss" and deletes the previous file - "A.scss" because it mistakes it for the "old sprite".
So basically it only allows one "SpritesmithPlugin" entry in the webpack config file now.

**Comment by mixtur** [comment]
That is weird. Different instances should not influence each other. Can you add your exact configuration please?

**Comment by liorch88** [comment]
Sure: 
```javascript
plugins: [
    new ExtractTextPlugin('[name].css'),
    new SpritesmithPlugin({
      src: {
        cwd: path.resolve(__dirname, 'src/widgets/forgot-password/assets/sprite-src'),
        glob: '*.png'
      },
      target: {
        image: path.resolve(__dirname, 'src/widgets/forgot-password/assets/sprite.png'),
        css: [[
          path.resolve(__dirname, 'src/widgets/forgot-password/styles/utils/sprite.scss')
        ]]
      },
    }),
    new SpritesmithPlugin({
      src: {
        cwd: path.resolve(__dirname, 'src/widgets/login/assets/sprite-src'),
        glob: '*.png'
      },
      target: {
        image: path.resolve(__dirname, 'src/widgets/login/assets/sprite.png'),
        css: [[
          path.resolve(__dirname, 'src/widgets/login/styles/utils/sprite.scss')
        ]]
      }
    }),
  ]
```

I've submitted a PR to fix this bug: #48

**Comment by mixtur** [comment]
Thank you for PR. It is so nice when your users are also programmers.

**Comment by Krasnopir** [comment]
if target image name have hash, after some changes in png icons new sprite image created with new hash, and we get two sprites with different hashes - old and new.



my way to delete extra files:

module.exports = {
    ...
    plugins: getPlugins(),
    ....
}

function getPlugins() {
    let plugins = [];
    clearOldFiles();
    plugins.push(
        pngSprite('main'),
        ....
    );
    return plugins;
}

const pngSprite = (folder, retina) => {
    const spritesmithOptions = {
        src: {
            cwd: path.resolve(__dirname, pngSpritesPath, folder),
            glob: '*.png'
        },
        target: {
            image: path.resolve(__dirname, pngSpritesPath, `${folder}_[hash:7].png`),
            css: [
                [
                    path.resolve(__dirname, sssSpritesPath, `${folder}.sss`),
                    { format: 'sugarss_handlebars' }
                ]
            ]
        },
        customTemplates: {
            'sugarss_handlebars': path.resolve(__dirname, sssPngSpritesTemplate),
            'sugarss_handlebars_retina': path.resolve(__dirname, sssPngSpritesRetinaTemplate)
        },
        apiOptions: {
            cssImageRef: `/image/png_sprites/${folder}_[hash:7].png`
        },
    };
    if(retina) spritesmithOptions.retina = '@2x';
    return(new SpritesmithPlugin(spritesmithOptions));
};


const clearOldFiles = () => {

    const filesFolders = [
        `${sssSpritesPath}`,
        `${pngSpritesPath}`,
    ];

    filesFolders.map(function (folder) {
        fs.readdirSync(folder).map(function (file) {
            if (fs.statSync(path.join(folder, file)).isFile()) {
                fs.unlinkSync(path.join(folder, file))
            }
        });
    });
};


So all files are deleted and then re-created for each bundle.
Also, if some sprites become unnecessary to us, for example, for a custom page, a landing page for a holiday, a new year, and so on - we can delete the source folder, but in my case only the actual files will remain.

For automatic generation of sprites from all folders of individual sprites.
I use

function getPlugins() {
    let plugins = [];

    clearOldFiles();

    const spritesPathList = fs.readdirSync(pngSpritesPath);
    spritesPathList.map(function (file) {
        return path.join(pngSpritesPath, file);
    }).filter(function (file) {
        return fs.statSync(file).isDirectory(file);
    }).forEach(function(el) {
        const spriteImagesList = fs.readdirSync(el);
        if (spriteImagesList.length > 0){
            const folderName = path.basename(el);
            plugins.push(pngSprite(folderName))
        }
    });

    return plugins;
}

**Comment by mixtur** [comment]
@Krasnopir what version are you using? webpack-spritesmith should already clean old files for you. On besteffortish basis, though. It cannot clean stuff after changing config. Or after previous run.

Still, I highly advice to use file-loader as in readme. You will still get hashes, and cleanup will become trivial if needed at all.
